### PR TITLE
[release/8.0] Seq component is broken with latest UseOtlpExporter - Convert Seq component to use OTEL's `AddProcessor` instead of `AddOtlpExporter` (#3697)

### LINKED ISSUE #3546: Seq component is broken with latest UseOtlpExporter (Does the `UseOtlpExporter` change prevent other Aspire components changing OTEL config?)
The [Seq Aspire component](https://learn.microsoft.com/en-us/dotnet/aspire/logging/seq-component?tabs=dotnet-cli) relies on the OTEL `AddOtlpExporter` method to extend telemetry configuration and this has now [been made incompatible with the ServiceDefaults generated code](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5400) (as `UseOtlpExporter` makes `AddOtlpExporter` throw an exception). 

Is there a way that an Aspire component can extend the telemetry configuration?

### PR DESCRIPTION
**Pull request by eerhardt**
Backport of #3697 to release/8.0

cc @liammclennan 

## Customer Impact

The Seq component no longer works with after the OTLP update to our ServiceDefaults project.

```
Unhandled exception. System.NotSupportedException: Signal-specific AddOtlpExporter methods and the cross-cutting UseOtlpExporter method being invoked on the same IServiceCollection is not supported.
```

## Testing

New automated tests added.
Manual testing that Seq works in the playground app.

## Risk

Low. The change only affects the Seq component.

## Regression?

Yes

_____

* Switch Seq OpenTelemetry configuration from `AddOtlpExporter` to `AddProcessor`.

* Switch Seq OpenTelemetry configuration from `AddOtlpExporter` to `AddProcessor`.

* Move OTLP exporter options onto SeqSettings.cs

* Add test

* Update README

* Renamed Seq settings

* Change order of config

* Updated code doc

* ConfigurationSchema.json

* Throw an exception if ServerUrl is unspecified, but HealthChecks is enabled.

* Fix tests

* PR feedback

Fix #3546
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3728)

### CODE REVIEW

**Comment by eerhardt** [comment]
This will also need https://github.com/dotnet/aspire/pull/3734 before it can be merged.

**Comment by eerhardt** [comment]
> This will also need #3734 before it can be merged.

I've updated this PR with these changes. It can now be merged once approved.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Need more approvals

**Comment by danmoseley** [comment]
Someone should sign off who has context on the change.

**Review by davidfowl — APPROVED** [review]

**Pull request merged**
